Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -288,6 +288,15 @@ May produce spurious warnings.
 May produce spurious warnings.") {
     IsHidden = true
   };
+  public static readonly Option<bool> SuggestProofRefactoring = new("--suggest-proof-refactoring", @"
+(experimental) Emits suggestions for moving assertions into by-proofs and hiding unused function definitions.
+May produce spurious suggestions. Use with --show-hints on the CLI.") {
+    IsHidden = true
+  };
+  public static readonly Option<bool> AnalyseProofs = new("--analyse-proofs", @"
+(experimental) Implies --warn-contradictory-assumptions, --warn-redundant-assumptions, and --suggest-proof-refactoring") {
```

Comment:
I think we use American English spelling, so this should be `analyze`.

Could you inline the documentation for those options (--warn-contradictory-assumptions etc..) here, instead of referring to them? Also, could you make those options hidden so they can be used for backwards compatibility, but `--analyse-proofs` replaces them. I think it's confusing to have both.

Also, consider adding options to toggle `off` the warnings when `--analyse-proofs` is used: `--allow-redundant-assumptions`, `--allow-contradictory-assumptions`